hack/check-workflows-uses-hashes.sh

```
@@ -0,0 +1,32 @@
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+#
+# Fails if any GitHub Actions workflow uses an external action without a full SHA pin.
+
+set -euo pipefail
+
+failed=0
+
+# Find all workflow YAML files
+for file in $(find .github/workflows/ -type f \( -name "*.yml" -o -name "*.yaml" \)); do
```

Shellcheck (0.10.0)

[warning] For loops over find output are fragile. Use find -exec or a while read loop.

(SC2044)